data-access-layer/iaas/AzureClient.js

```diff
@@ -27,6 +27,58 @@ class AzureClient extends BaseCloudClient {
       .value()
     );
     Promise.promisifyAll(this.computeClient.snapshots);
+    Promise.promisifyAll(this.computeClient.disks);
+  }
+
+  getSnapshot(snapshotId) {
+    return this.computeClient.snapshots.getAsync(this.settings.resource_group, snapshotId)
+      .then(snapshotResponse => _.pick(snapshotResponse, ['location', 'creationData', 'id']));
+  }
+
+  createDiskFromSnapshot(snapshotId, zones, opts = {}) {
+    const diskName = this.getRandomDiskId();
+    return Promise.try(() => this.getSnapshot(snapshotId))
+      .then(snapshotData => {
+        const options = {
+          zones: _.isArray(zones) ? zones : [zones],
+          location: snapshotData.location,
+          tags: _.assign({}, opts.tags || {}, {
+            createdBy: 'service-fabrik'
+          }),
+          sku: _.assign({}, opts.sku || {}, {
+            name: 'Premium_LRS',
+            tier: 'Premium'
+          }),
+          creationData: {
+            createOption: 'Copy',
```

Comment:
Since this method is for creating a disk from snapshot, the default create option has to be always "Copy" (at least till we do not use a different api version of Azure). I can move it to a constant at the utmost